parlai/core/params.py

```diff
@@ -332,6 +333,16 @@ def add_extra_args(self, args=None):
         model = parsed.get('model', None)
         if model is not None:
             self.add_model_subargs(model)
+        else:
+            # try to get model name from model opt file
+            model_file = parsed.get('model_file', None)
+            if model_file is not None:
+                optfile = model_file + '.opt'
+                if os.path.isfile(optfile):
+                    with open(optfile, 'rb') as handle:
+                        new_opt = pickle.load(handle)
+                        if 'model' in new_opt:
+                            self.add_model_subargs(new_opt['model'])
```

Comment:
can you double-check this doesn't kill the convai2 eval_ppl? it might get angry since we try to load the model params into a subclass of the model

Reply:
ah good call -- checked and it doesn't break! in this case, 'model' should already be set so it will not load it from the opt file